### ISSUE TITLE
[question] Is it possible to read the entire file-system from the executable?

### ISSUE THREAD

**Issue by kittaakos** [nexe/nexe#613]
**Is this a BUG or a FEATURE REQUEST?:** 
A question, maybe a bug.

Hi there,

I am using `nexe` to bundle my application into a "single" binary. ❤️

I have managed to get it running, but I cannot see/list the entire file-system correctly from my application started from the binary. Is this a known limitation, a possible bug or I have messed up something with the configuration.

I have seen the documentation for the `nexe-fs`, but freely speaking, it was not clear to me how to use it, if I have to use it at all.

My repository is here: https://github.com/kittaakos/theia-nexe
If you want to try it in Gitpod: https://gitpod.io/#https://github.com/kittaakos/theia-nexe
Or you can get an executable for Linux and macOS from here: https://dev.azure.com/kittaakos/theia-sandbox/_build/results?buildId=230

Thank you!

----

PS:
> I cannot see/list the entire file-system correctly

If this is unclear, but `nexe` supposed to fully support FS read/write, I am happy to describe the oddities I have experienced.

**Comment by paulocoghi** [comment]
Can you test again using v3.2.0?

**Comment by kittaakos** [comment]
@paulocoghi, thank you for the update. I have quickly glanced into the behavior with `3.2.0`, it looks promissing, I need to verify it further, though.

**Comment by paulocoghi** [comment]
Kudos to @calebboyd :tada:

**Comment by calebboyd** [comment]
Oh hey thanks,

Note -- I didn't explicitly work to solve this issue in 3.2.0. So I don't actually think it is fixed.

Related issues: #571 #618

**Comment by kittaakos** [comment]
> Can you test again using v3.2.0?

I have checked it again with `nexe@^3.2.0`, and I have the same issue. For instance, when I start the application I cannot list the files, directories under `~/Desktop`. By default, we use `fs-extra` instead of `fs`, could that be the cause of my issue?. Do I have to shim `fs-extra`? Thanks!

**Comment by ricardopolo** [comment]
I am not able to list files in the root of the compiled binary. 

@calebboyd  If the issue is fixed in 3.2.0, this should work now, or do we need to wait for you to release new precompiled versions? Do you have any ETA for this?

Thank you!!

**Comment by paulocoghi** [comment]
@ricardopolo, I had mistakenly understood that the correction made by @calebboyd also solved this issue, but it's not the case.

**Comment by ArashMotamedi** [comment]
`fs.promises.readdir` works

**Comment by a1994846931931** [comment]
@kittaakos Hello Kitta! I am trying to bundle theia as a single binary recently as well. It seems that you have dug a log into this some a year ago. May I ask, have you managed to achieve that with nexe at the end?

**Comment by kittaakos** [comment]
> have you managed to achieve that with nexe at the end?

No, I have not. But I have not re-evaluated it since https://github.com/nexe/nexe/issues/613#issuecomment-488577972.
